src/agent/src/image_rpc.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::fs;
+use std::process::{ExitStatus, Command};
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use protocols::image;
+use tokio::sync::Mutex;
+use ttrpc::{self, error::get_rpc_status as ttrpc_error};
+
+use crate::sandbox::Sandbox;
+//use crate::trace_rpc_call;
+use crate::rpc::CONTAINER_BASE;
+
+const SKOPEO_PATH: &str = "/usr/bin/skopeo";
+const UMOCI_PATH: &str = "/usr/local/bin/umoci";
+
+// Convenience macro to obtain the scope logger
+macro_rules! sl {
+    () => {
+        slog_scope::logger()
+    };
+}
+
+pub struct ImageService {
+    sandbox: Arc<Mutex<Sandbox>>,
+}
+
+impl ImageService {
+    pub fn new(sandbox: Arc<Mutex<Sandbox>>) -> Self {
+        Self { sandbox }
+    }
+
+    fn pull_image_from_registry(image: &str, cid: &str, source_creds: &Option<&str>) -> Result<()> {
+        let source_image = format!("{}{}", "docker://", image);
```

Comment:
Why are we hard-coding `docker` here?

Reply:
This should be the way scpo is used, as I understand it. Here is the help message it prints.
> Supported transports:
> containers-storage, dir, docker, docker-archive, docker-daemon, oci, oci-archive, ostree, tarball

Reply:
Yes, skopeo uses the `docker:` prefix to indicate a container registry which could be anything. Few examples here will make it clearer -  https://github.com/containers/skopeo/blob/main/docs/skopeo-copy.1.md